src/lib/consenthandler/consentHandler.js

```diff
@@ -41,20 +41,28 @@ export default class ConsentHandler{
      * @returns {boolean}
      */
     consentEnabled(){
-        return !!this.proxy;
+        return this.config.type === 'iab';
     }
 
     /**
      * Test whether the publisher site has consent to access privacy data
      * @param callback
      */
     hasSiteConsent(callback){
-        this.checkConsent((consent,success)=>{
-            let hasConsent = false;
-            if(success){
-                hasConsent = consent.hasSiteConsent;
+        this.checkConsent((consentData)=>{
+            let gdprApplies = consentData.gdprApplies;
+            if (gdprApplies === undefined) {
+                // If alwaysCallback is true, then when gdprApplies is ambiguous, treat it as false
+                gdprApplies = !this.config.alwaysCallback;
+            }
+
+            if (gdprApplies) {
+                const hasStorageAccess = (typeof consentData.hasStorageAccess === 'function') ? consentData.hasStorageAccess() : false;
```

Comment:
why return a function that will give the same result every time instead of storing the result once and return it whenever hasSiteConsent is called

Reply:
Yeah, that could work, too.  It comes down to not needing to eval the value if not used.

Reply:
I'll add a flag to replace the functions.